### PR TITLE
Document wildcard for network interfaces in Network Settings

### PR DESCRIPTION
**Pull request by GoodMirek**
This issue was supposed to be fixed already, as mentioned in already closed https://github.com/elastic/elasticsearch/issues/14607. This is actually its duplicate.

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
Yes
- Have you followed the [contributor guidelines](https://github.com/elastic/elasticsearch/blob/master/CONTRIBUTING.md)?
I believe so
- If submitting code, have you built your formula locally prior to submission with `gradle check`?
NA
- If submitting code, is your pull request against master? Unless there is a good reason otherwise, we prefer pull requests against master and will backport as needed.
NA
- If submitting code, have you checked that your submission is for an [OS that we support](https://www.elastic.co/support/matrix#show_os)?
NA
- If you are submitting this code for a class then read our [policy](https://github.com/elastic/elasticsearch/blob/master/CONTRIBUTING.md#contributing-as-part-of-a-class) for that.
NA

### CODE REVIEW

**Comment by javanna** [comment]
cc @elastic/es-core-infra

**Comment by javanna** [comment]
ping @elastic/es-core-infra can we merge this one?

**Review by danielmitterdorfer — CHANGES_REQUESTED** [review]
Thanks for your PR and I am very sorry that this has fallen through the cracks for so long.

I can see the intent behind your change but I'm not sure whether this is the right place to document this. Yes, `0.0.0.0` is a specialized address according to [RFC 5735](https://tools.ietf.org/html/rfc5735) but in that part of our documentation we talk about values that are treated specially by Elasticsearch itself (outside of any standard). Furthermore, in the following section we refer to these special values again and mention that one can add `:ipv4` and `:ip6` suffixes to special values which is not possible for `0` (or `0.0.0.0`). I think what would work though is to mention that `0.0.0.0` is a possible value in the section `network.host`?

**Comment by GoodMirek** [comment]
@danielmitterdorfer I hope I have understood you well. I have rebased and made changes.

I have moved `0.0.0.0` to the section `network.host`, as recommended in the review. I left `0` among special values, as far as I understand it is a special value. I have added a comment related to the applicability of specifiers to this special value, as mentioned in the review.
I have not personally tested or reviewed in the source code that specifiers `:ipv4` and `:ipv6` cannot be used with the special value `0`; I rely upon the review.

File: docs/reference/modules/network.asciidoc [hunk 1]
```diff
@@ -80,6 +80,9 @@ The following special values may be passed to `network.host`:
 `_global_`::
 
   Any globally-scoped addresses on the system, for example `8.8.8.8`.
+  
+  `0`::
```
Comment (danielmitterdorfer) [inline docs/reference/modules/network.asciidoc]: The fact that `0` is recognized at all is due to [IP address normalization in OpenJDK](http://hg.openjdk.java.net/jdk/jdk/file/190b77982361/src/java.base/share/classes/sun/net/util/IPAddressUtil.java#l41) and I don't think that we should document that behavior as a "special value". Special values documented here are meant as "specially treated by Elasticsearch itself", i.e. those have a  [specific handling in the `NetworkService`](https://github.com/elastic/elasticsearch/blob/a9641960e1be101c0bcd4e9bbd28c3a2a2cabeb5/server/src/main/java/org/elasticsearch/common/network/NetworkService.java#L209-L226).

File: docs/reference/modules/network.asciidoc [hunk 2]
```diff
@@ -24,7 +24,7 @@ host to other nodes in the cluster. Accepts an IP address, hostname, a
 <<network-interface-values,special value>>, or an array of any combination of
 these. Note that any values containing a `:` (e.g., an IPv6 address or
 containing one of the <<network-interface-values,special value>>) must be quoted
-because `:` is a special character in YAML.
+because `:` is a special character in YAML. `0.0.0.0` is an acceptable IP address, it has the same effect as the special value `0`.
```
Comment (danielmitterdorfer) [inline docs/reference/modules/network.asciidoc]: I think this is fine. As I still would not document `0` as "special value" (see rationale below), I think you could merge the text to say something along the lines of:

> `0.0.0.0` is an acceptable IP address and will binds to all network interfaces.

Can you please also wrap text after 80 characters in docs?

**Review by danielmitterdorfer — CHANGES_REQUESTED** [review]
Thanks for iterating @GoodMirek. The change makes sense to me. I still think we should not document `0` as special value and included a rationale inline as well as one more suggestion for the docs. Can you please have a look? I think we need another iteration but are close to merging it.

File: docs/reference/modules/network.asciidoc (hunk 2 above)
Comment (GoodMirek) [reply docs/reference/modules/network.asciidoc]: My primary motivation for this PR is to document the value `0`, as I found that in some config examples and had hard time to find what that means.

Do you agree with:
> `0.0.0.0` is an acceptable IP address and will bind to all network interfaces. Value `0` has the same effect as value `0.0.0.0`.

?

Comment (danielmitterdorfer) [reply docs/reference/modules/network.asciidoc]: Yes, I am ok with this change. Just a very minor nit; the articles are missing, i.e.

> The value `0` has the same effect as the value `0.0.0.0`.

Comment (danielmitterdorfer) [reply docs/reference/modules/network.asciidoc]: @GoodMirek I think we're very close to completion now. Are you still interested in getting your change in? :)

Comment (GoodMirek) [reply docs/reference/modules/network.asciidoc]: @danielmitterdorfer Will look next week. What is a process for merging these changes to current master or whatever the development branch is?
Will I have to go through the whole process again?

Comment (danielmitterdorfer) [reply docs/reference/modules/network.asciidoc]: Thanks @GoodMirek. There is no need to concern yourself with other branches than this one. I will take care of merging this to the correct branches on your behalf via cherry-picking (so you will still show up as the author of the change).

Comment (danielmitterdorfer) [reply docs/reference/modules/network.asciidoc]: @GoodMirek Any chance you can push the commit including our final changes? Then I can take it from there.

**Review by danielmitterdorfer — APPROVED** [review]
Thanks for the PR! I went ahead and merged the latest `6.2` branch and also did the changes we mentioned during the review so we're able to merge this. LGTM

**Pull request merged**